src/servers/Http1Server.h

```diff
@@ -59,6 +61,9 @@ class Server: public ConnStateData
     Http1::RequestParserPointer parser_;
     HttpRequestMethod method_; ///< parsed HTTP method
 
+    /// whether the client requested to upgrade
+    bool mayUpgrade = false;
```

Comment:
> Does any Factory code use scc.reason already? If not, I suggest removing that unused field. It can always be added later if needed.

Removed. However still I believe that it was useful just to mark that this class can be useful for other "reasons" than upgrading to new proto. It was not hurting anything in any case.

>... if the description is correct, then we should use a much more precise name:
Yes it is correct. Renamed

Reply:
The two requests addressed at 41fc1c5a